### ISSUE TITLE
[MLIR] MSVC build failure: usage of std::complex<llvm::APFloat> is unspecified.

### ISSUE THREAD

**Issue by aganea** [llvm/llvm-project#61714]
Building MLIR using the latest MSVC on Windows, emits a bunch of warnings such as:
```console
[8166/8337] Building CXX object tools\mlir\test\lib\Dialect\Test\CMakeFiles\MLIRTestDialect.dir\TestDialect.cpp.obj
C:\git\llvm-project\mlir\include\mlir/IR/BuiltinAttributes.h(353): warning C4996: 'std::complex<llvm::APFloat>::complex': warning STL4037: The effect of instantiating the template std::complex for any type other than float, double, or long double is unspecified. You can define _SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING to suppress this warning.
```
Using:
```bat
cmake -GNinja %ROOT%/llvm -DCMAKE_BUILD_TYPE=Debug -DLLVM_OPTIMIZED_TABLEGEN=ON -DLLVM_ENABLE_LIBXML2=OFF -DCMAKE_C_COMPILER=cl.exe -DCMAKE_CXX_COMPILER=cl.exe -DCMAKE_LINKER=link.exe -DLLVM_ENABLE_PROJECTS="llvm;clang;lld;clang-tools-extra;mlir" -DLLVM_USE_CRT_DEBUG=MTd -DLLVM_ENABLE_PDB=ON -DLLVM_ENABLE_ASSERTIONS=ON
```
[Latest WG21 working draft](https://www.open-std.org/JTC1/SC22/WG21/docs/papers/2023/n4928.pdf) says:

> _28.4.1/2/ The effect of instantiating the template complex for any type that is not a cv-unqualified floating-point type (6.8.2) is unspecified._

**Comment by joker-eph** [comment]
The warning seems legit, isn't it something that we should fix?

**Comment by joker-eph** [comment]
Let keep the discussion in the duplicate #65255
